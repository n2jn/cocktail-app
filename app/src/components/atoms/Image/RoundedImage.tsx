import React from 'react';
import {Image, ImageProps, StyleSheet, View} from 'react-native';
import {useImageProps} from './hooks/useImageProps';

type AcceptableImageSize = 8 | 16 | 24 | 32 | 64;

type RoundedImageProps = Omit<ImageProps, 'source'> & {
  image: string;
  size: AcceptableImageSize;
};

export const RoundedImage = ({
  image,
  size,
  ...imageProps
}: RoundedImageProps) => {
  const {onError, source, style} = useImageProps(image, size);

  return (
    <View style={styles.container}>
      <Image
        resizeMode="contain"
        testID="image"
        source={source}
        onError={onError}
        style={style}
        {...imageProps}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    borderRadius: 20,
    backgroundColor: 'black',
    overflow: 'hidden',
  },
});
