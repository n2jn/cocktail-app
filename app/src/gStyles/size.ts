import {Dimensions} from 'react-native';

// https://medium.com/react-native-training/scaling-react-native-apps-for-tablets-211de8399cf1

const guidelineBaseWidth = 320;
const guidelineBaseHeight = 680;

// will be used for width !
export const scale = (size: number) => {
  const {width} = Dimensions.get('window');
  return Math.round((width / guidelineBaseWidth) * size);
};

// will be used for height !
export const verticalScale = (size: number) => {
  const {height} = Dimensions.get('window');
  return Math.round((height / guidelineBaseHeight) * size);
};

// will be used for padding / margin !
export const moderateScale = (size: number, factor: number = 0.5) =>
  Math.round(size + (scale(size) - size) * factor);

export const scaled64 = scale(64);
export const scaled32 = scale(32);
export const scaled24 = scale(24);
export const scaled16 = scale(16);
export const scaled12 = scale(12);
export const scaled8 = scale(8);
