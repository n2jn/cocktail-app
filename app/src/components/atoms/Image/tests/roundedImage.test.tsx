import {fireEvent, render, screen} from '@testing-library/react-native';
import {scale} from '~gStyles/size';
import {RoundedImage} from '../RoundedImage';

const renderComponent = () => {
  const image = '';
  const size = 8;

  render(<RoundedImage image={image} size={size} />);
  return {
    image,
    size,
  };
};

describe('Image with error', () => {
  test('Should have placeholder image', async () => {
    const {size} = renderComponent();
    const source = await screen.findByTestId('image');
    fireEvent(source, 'error');
    const scaledSize = scale(size);
    expect(source).toHaveProp('source', {
      uri: `https://dummyimage.com/${scaledSize}x${scaledSize}/000/ffffff.jpg&text=??`,
    });
  });
});
