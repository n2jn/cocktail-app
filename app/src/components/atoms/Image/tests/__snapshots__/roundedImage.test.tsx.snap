// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic test Button renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "black",
      "borderRadius": 20,
      "overflow": "hidden",
    }
  }
>
  <Image
    onError={[Function]}
    resizeMode="contain"
    source={
      Object {
        "uri": "",
      }
    }
    style={
      Object {
        "height": 19,
        "width": 19,
      }
    }
    testID="image"
  />
</View>
`;
